jerry-core/ecma/builtin-objects/ecma-builtin-internal-routines-template.inc.h

```diff
@@ -39,6 +39,7 @@
 #define ROUTINE_ARG_LIST_1 ROUTINE_ARG_LIST_0 ROUTINE_ARG(1)
 #define ROUTINE_ARG_LIST_2 ROUTINE_ARG_LIST_1 ROUTINE_ARG(2)
 #define ROUTINE_ARG_LIST_3 ROUTINE_ARG_LIST_2 ROUTINE_ARG(3)
+#define ROUTINE_ARG_LIST_4 ROUTINE_ARG_LIST_3 ROUTINE_ARG(4)
```

Comment:
Instead of introducing these new 4 length routine args, how about writing a custom dispatcher for the object? As I saw most of the time the `this_arg` is unused.